test/numerical/Common.hpp

```diff
@@ -1,9 +1,12 @@
-/*
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
-#include <rapidcheck.h>
-#undef _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
-
-#include "test/modellib/ModelLib.hpp"
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"   // gcc, clang
+#define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS // MSVC
```

Comment:
@sstamenova I believe this fix from PR #1626 is MSVC specific, it didn't work for me with clang on Mac

Reply:
So the `#pragma GCC diagnostic ignored "-Wdeprecated"` is the equivalent of  this flag?
And does this works on MSVC? if so, let's have as little redundancy as possible, tx

Reply:
Since MSVC != GCC, the GCC diagnostics don't impact the MSVC builds. I suggest using an `ifdef` or similar to check which compiler it is and then include the appropriate suppression. For example: `#if defined(__GNUC__)`, `if defined(__clang__)`, `if defined(_MSC_VER)`, etc.

You can always look at the Windows CI build to see whether there are any warnings/errors.

Reply:
thanks for the suggestion, I now choose between MSVC and gcc/clang with `#if defined(_MSC_VER)`

I verified that `#pragma GCC` works for clang, which is also explained here: https://stackoverflow.com/a/11853015